third_party/move/move-vm/runtime/src/loader/modules.rs

```diff
@@ -274,9 +284,9 @@ impl Module {
         natives: &NativeFunctions,
         size: usize,
         module: Arc<CompiledModule>,
-        cache: &ModuleStorageAdapter,
-        name_cache: &StructNameCache,
-    ) -> Result<Self, (PartialVMError, Arc<CompiledModule>)> {
+        struct_ability_checker: &impl StructTypeAbilityChecker,
```

Comment:
nice way to unify the API between V1/V2! 

wouldn't another way be to provide a trait that gets a struct type (looking at both implementations in struct_type_ability_checker), then call check_compatibility here (in the caller)? Wouldn't be as paranoid mode specific and could be worth if if there is any use for more generality (of getting struct_ty)

Reply:
Indeed! Changed the trait to fetch struct types instead.